### PR TITLE
[Android] Follow-on to keyboard/model update notifications

### LINKED ISSUE #2367: [Android] UX to give users more control on background notifications
With the background notifications implemented in #2335, if a user "Cancels" a notification, Keyman will still check another day and the notification will come back.

@mcdurdin noted:
> This is probably a blocker for this PR. Notifications that keep coming back are extremely annoying to the user. If they dismiss the notification, we need to remember that they ignored that update and not pester them about it again. For a new update, then the banner can appear again.
>
> That reminds me, do we have a way to turn notifications and/or update checks on/off entirely?

### PR DESCRIPTION
**Pull request by darcywong00**
Follow-on to #2412 and fixes #2367

After the user "Cancels" an update notification, the preference to ignore that update for 3 months is saved.

> Although if they open the keyboard details in the Settings menu, that might be a good place to note it has an update available, but not top level 'nagging' style

This PR updates the "Keyboard Settings page" and "Model Info page" so if a newer version from the cloud exists, a right arrow is available for the user to download the update. The currently installed version is displayed, and clicking on the version row brings up the resource download dialog.

### SIL EuroLatin Update Available
sil eurolatin update

### Screenshot en.mtnt Update Available
en mtnt model update

Note: These screenshots were generated by a development build where Keyman initializes with sil-eurolatin-1.8.js and version 0.1.2 of en.mtnt model.

### CODE REVIEW

File: android/KMEA/app/src/main/java/com/tavultesoft/kmea/ModelPickerActivity.java [hunk 1]
```diff
@@ -82,10 +82,10 @@ protected void onResume() {
     super.onResume();
     if (!didExecuteParser) {
       didExecuteParser = true;
-      //new JSONParse().execute();
       repo = CloudRepository.shared.fetchDataset(context);
 
-      listView.setAdapter(new FilteredLexicalModelAdapter(context, repo, languageID));
+      // Initialize the dataset of installed lexical models
+      listView.setAdapter(new FilteredLexicalModelAdapter(context, KeyboardPickerActivity.getInstalledDataset(context), languageID));
```
Comment (darcywong00) [inline android/KMEA/app/src/main/java/com/tavultesoft/kmea/ModelPickerActivity.java]: This change needed so Model Info page will correctly show the installed versions.

**Comment by mcdurdin** [comment]
I'm not quite sure I understand the workflow here. I can't see any notification of an available update in the screenshots?

**Comment by darcywong00** [comment]
The right arrows convey that the user can click on the row to do something.
In the "Settings" menu, the right arrow on Languages will advance to the available keyboards
On the help links, the right arrow will bring up the corresponding help page.

I thought about making a new row for "Install update"
but it may make the page too long w/ the QR code. Other keyboards will also have a row for "Uninstall keyboard" (not displayed on SIL EuroLatin)

**Comment by darcywong00** [comment]
Here's a screenshot using the cloud download icon

cloud update avail

**Comment by mcdurdin** [comment]
Could you change the text under "Dictionary version" from "0.1.3" to "0.1.3 (Update available!)" to make it clearer to the user why they should touch that icon?

**Comment by darcywong00** [comment]
Hmm, the bigger icon is causing the QR code caption to get cropped. I might have to make the Keyboard Settings page scroll-able...

sil eurolatin cloud update

**Comment by mcdurdin** [comment]
> Hmm, the bigger icon is causing the QR code caption to get cropped. I might have to make the Keyboard Settings page scroll-able...

Given the variety of screen sizes available, I think this is pretty essential.

**Comment by darcywong00** [comment]
> Given the variety of screen sizes available, I think this is pretty essential.

OK, it took some wrangling with the layouts, but I've got it where the QR code gets appended to the scroll-able listview. I also was able to clean up some unused bottom padding.

**Comment by mcdurdin** [comment]
Quick question: does this ping the cloud server every time you open the menu? What happens if offline? Just can't see that pathway clearly

**Comment by darcywong00** [comment]
> Quick question: does this ping the cloud server every time you open the menu?

The menus are accessing the stored & merged list of cloud and locally installed resources. It's the Dataset backing that @jahorton had refactored.

> What happens if offline?

I tested offline:
The menus display the same ("Update Available" and the cloud download icon). 
Clicking on the icon brings up the usual confirmation dialog. 
And then clicking the "DOWNLOAD" confirmation brings up the "No internet connection" toast notifiaction.

**Pull request merged**